apps/src/assetManagement/IconList.jsx

```diff
@@ -0,0 +1,84 @@
+var IconListEntry = require('./IconListEntry.jsx');
+var icons = require('./icons').aliases;
+var msg = require('../locale');
+
+/**
+ * A component for managing icons.
+ */
+module.exports = React.createClass({
+  propTypes: {
+    assetChosen: React.PropTypes.func.isRequired,
+    search: React.PropTypes.string.isRequired
+  },
+
+  getMatches: function (query) {
+    var letters = query.split(''), results = {};
+
+    // Walk the prefix tree to find matches
+    for (var i = 0, node = icons; i < letters.length; i++) {
+      node = node[letters[i]];
+      if (!node) {
+        // No matches found
+        break;
+      }
+    }
+
+    // Iterate leaf nodes to construct the results
+    function getResults(node, prefix) {
+      if (node.$) {
```

Comment:
$ was shorthand for a leaf node (it can't be an alphanumeric character) but I've removed this complexity in 9fc672c9ea4499b83ca039cc19a5697e97f3e0fd.